SDCTA/src/error_handling/auth-errors.tsx

```diff
@@ -0,0 +1,57 @@
+const signUpErrorHandler = async (error: Error) => {
```

Comment:
A slight variation on the InputError type I talked about above:
You could have these handlers return an InputError type instead of a tuple. You can do the same checks you're doing now in these functions but just set the correct key in the InputError object you're going to return when you find an error. Then return that InputError object. If there are no errors, the InputError object will just have keys with blank strings. Then instead of:

            setPasswordError("");
            setEmailError("");
            setUnknownError("");
            if (error instanceof Error) {
                const errorMessage = await logInErrorHandler(error);
                if (errorMessage[0] === "email") {
                    setEmailError(errorMessage[1]);
                }
                if (errorMessage[0] === "password") {
                    setPasswordError(errorMessage[1]);
                }
                if (errorMessage[0] === "unknown") {
                    setUnknownError(errorMessage[1] + " (Reload and try again)");
                }
            }

You can do something like:
if (error instanceOf Error) {
    setInputError(loginErrorHandler(error));
}

This is kind of rough off the top of my head, will prob have to account for some details when you implement it